### PR TITLE
[perf] add cuBLASLt NVFP4 GEMM kernel to sgl fp4 gemm backend

### PR DESCRIPTION
**Pull request by Qiaolin-Yu**
## Accuracy Tests
Tested on Kimi K2.5 NVFP4, gpqa 87%, have 1%~2% e2e improvement than flashinfer_cutlass

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

**Comment by DarkSharpness** [comment]
Currently `tvm_ffi` does not support directly access the cublas handle from torch, so we may have to create a new one. I'm not sure about the extra resource usage here. If heavy, can we think of some way to get torch native cublas handle?

**Review by b8zhong — APPROVED** [review]
Looks fine. Btw, you can use the Torch scaled MM to achieve same thing

Also we recently switch to Cute-DSL as default NVPF4 GEMM backend, since I find it can bring speedup in most scenarios. Not sure if you want to recompare

```
import torch
  ST, SW = torch._C._ScalingType, torch._C._SwizzleType
  # A:[M,K/2] B:[N,K/2] packed fp4 (torch.float4_e2m1fn_x2)
  # asf/bsf: [M,K/16]/[N,K/16] float8_e4m3fn, swizzled to 32_4_4
  # ga/gb: fp32 scalar global scales (alpha == ga*gb)
  out = torch._scaled_mm_v2(
      a, b.t(),
      [asf, ga], [int(ST.BlockWise1x16.value), int(ST.TensorWise.value)],
      [int(SW.SWIZZLE_32_4_4.value), int(SW.NO_SWIZZLE.value)],
      [bsf, gb], [int(ST.BlockWise1x16.value), int(ST.TensorWise.value)],
      [int(SW.SWIZZLE_32_4_4.value), int(SW.NO_SWIZZLE.value)],
      None, torch.bfloat16,
  )
```
